zyte_spider_templates/pages/search_request_template.py

```diff
@@ -0,0 +1,318 @@
+            if len(form_excerpt) >= 64:
+                form_excerpt = form_excerpt[:-1] + "…"
+            raise ValueError(
+                f"Did not find an input field for the search query in "
+                f"the most likely search form at {self.response.url} "
+                f"(form_excerpt)."
+            )
+        try:
+            request_data = form2request(form, data, click=submit_button)
+        except NotImplementedError:
+            raise ValueError("form2request does not support the target search form")
+        return SearchRequestTemplate(
+            url=request_data.url.replace(_PLACEHOLDER, "{{ query|quote_plus }}"),
+            method=request_data.method,
+            headers=request_data.headers,
+            body=request_data.body.decode().replace(
+                _PLACEHOLDER, "{{ query|quote_plus }}"
+            ),
+        )
+
+    @validates_input
+    async def to_item(self) -> SearchRequestTemplate:
+        builders = {
+            "extruct": self._item_from_extruct,
+            "formasaurus": self._item_from_formasaurus,
+            "link_heuristics": self._item_from_link_heuristics,
+            "form_heuristics": self._item_from_form_heuristics,
+        }
+        builder_ids = self.page_params.get("search_request_builders", list(builders))
+        builder_strategy = self.page_params.get(
+            "search_request_builder_strategy", "popular"
+        )
+        if builder_strategy not in {"first", "popular"}:
+            raise ValueError(
+                f"Unsupported search_request_builder_strategy value: {builder_strategy!r}"
+            )
+        results = defaultdict(list)
+        for builder_id in builder_ids:
+            builder = builders[builder_id]
+            try:
+                result = builder()
+            except ValueError:
+                continue
+            if result:
+                if builder_strategy == "first":
+                    return result
+                results[(result.url, result.body)].append((builder_id, result))
+        if results:
+            assert builder_strategy == "popular"
+            top_count = max(len(v) for v in results.values())
+            top_results = {
+                builder_id: result
+                for result_list in results.values()
+                for builder_id, result in result_list
+                if len(result_list) == top_count
+            }
+            for builder_id in builder_ids:
+                if builder_id not in top_results:
+                    continue
+                return top_results[builder_id]
+
+        if not isinstance(self.response.response, BrowserResponse):
+            logger.error(
+                f"Cannot build a search request template for "
+                f"{self.response.url}. Try enabling browser rendering by "
+                f'setting "browserHtml" to True in the '
+                f"ZYTE_API_PROVIDER_PARAMS setting dict."
+            )
```

Comment:
Is it a common error you've faced during the tests?

Reply:
With the current implementation, during tests, 3194 domains worked with browser, while only 3113 worked with HTTP, so I guess you could say it is a real issue, it maybe not that significant.

If you are thinking that maybe we don’t need a browser-specific issue, and we should just suggest people to write a page object if things fail, I am fine with that.

Reply:
I don't know! For the UI users, the message could be confusing, as the solution is not to change ZYTE_API_PROVIDER_PARAMS (it's not clear how to do it, and I'm not even sure it'd work), but to change the spider extract_from argument. The easiest seems to be to drop this message :)